# Quiz Questions

In [2]:
import numpy as np

In [3]:
import numpy as np
import pandas as pd
from pandas import DataFrame

np.random.seed(12)

n = 100

In [4]:
# Seed the pseudo-random number generator
np.random.seed(12)

# Sample size
n=100

# Create data
wcc = np.round(np.random.normal(15, 5, n), 0)
crp = (wcc * 2) + np.round(np.random.normal(0, 10, n), 0)

lipase = wcc + crp + np.round(np.random.normal(2, 10, n), 0)

# Pandas dataframe obejct
df = DataFrame(
    {'WCC':wcc,
     'CRP':crp,
     'Lipase':lipase}
)

In [9]:
!pip install plotly

zsh:1: command not found: pip


In [11]:
import sys
!{sys.executable} -m pip install plotly

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.9/9.9 MB 14.9 MB/s  0:00:00 eta 0:00:01
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2/2 [plotly]2m1/2 [plotly]

[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: pip3 install --upgrade pip


In [15]:
import sys
!{sys.executable} -m pip install nbformat ipython

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7/7 [nbformat]

[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: pip3 install --upgrade pip


In [17]:
import plotly.io as pio
pio.renderers.default = "browser"

px.scatter_matrix(df, title='Scatter plot matrix')

In [18]:
import plotly.io as pio
pio.renderers.default = 'vscode'

In [14]:
import pandas as pd

df = pd.read_csv('/Users/suchipatel/Downloads/Students Social Media Addiction.csv')

In [15]:
px.scatter_matrix(
    df,
    title='Scatter plot matrix'
)

In [19]:
fig = px.scatter_matrix(df, title='Scatter plot matrix')
fig.show()

In [21]:
from patsy import dmatrices

In [23]:
pip install patsy


[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: pip3 install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [26]:
print(list(df.columns))

['Student_ID', 'Age', 'Gender', 'Academic_Level', 'Country', 'Avg_Daily_Usage_Hours', 'Most_Used_Platform', 'Affects_Academic_Performance', 'Sleep_Hours_Per_Night', 'Mental_Health_Score', 'Relationship_Status', 'Conflicts_Over_Social_Media', 'Addicted_Score']


In [28]:
df.columns = df.columns.str.strip()
print(df.columns)

Index(['Student_ID', 'Age', 'Gender', 'Academic_Level', 'Country',
       'Avg_Daily_Usage_Hours', 'Most_Used_Platform',
       'Affects_Academic_Performance', 'Sleep_Hours_Per_Night',
       'Mental_Health_Score', 'Relationship_Status',
       'Conflicts_Over_Social_Media', 'Addicted_Score'],
      dtype='str')


In [33]:
[col for col in df.columns if 'crp' in col.lower()]

[]

In [34]:
df.columns = df.columns.str.strip()
[col for col in df.columns if 'crp' in col.lower()]

[]

In [37]:
# Target vector and design matrix
y, X = dmatrices('Addicted_Score ~ Avg_Daily_Usage_Hours + Sleep_Hours_Per_Night', data=df)

### 1.Create multiple linear regression model with WCC and CRP as predictors of Lipase.

In [38]:
import statsmodels.api as sm
from patsy import dmatrices

y, X = dmatrices(
    'Mental_Health_Score ~ Avg_Daily_Usage_Hours + Sleep_Hours_Per_Night',
    data=df
)

model = sm.OLS(y, X).fit()
print(model.summary())

                             OLS Regression Results                            
Dep. Variable:     Mental_Health_Score   R-squared:                       0.656
Model:                             OLS   Adj. R-squared:                  0.655
Method:                  Least Squares   F-statistic:                     670.4
Date:                 Tue, 07 Apr 2026   Prob (F-statistic):          1.50e-163
Time:                         20:36:06   Log-Likelihood:                -693.76
No. Observations:                  705   AIC:                             1394.
Df Residuals:                      702   BIC:                             1407.
Df Model:                            2                                         
Covariance Type:             nonrobust                                         
                            coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------------
Intercept           

#### A multiple linear regression model was created to examine whether average daily usage hours and sleep hours per night predict mental health score.

### 2. Comment on the F statistic.

#### The F-statistic is 670.4 with a p-value of 1.50e-163, which is far below 0.05. This indicates that the overall model is statistically significant, meaning that the predictors jointly explain a significant portion of the variation in mental health score.

### 3. Comment on the individual coefficients and their p values.

#### The coefficient for Avg_Daily_Usage_Hours is -0.5666 and is statistically significant (p < 0.001). This suggests that increased social media usage is associated with a decrease in mental health score. The coefficient for Sleep_Hours_Per_Night is 0.1939 and is also statistically significant (p < 0.001). This indicates that more sleep is associated with higher mental health scores. Both predictors are statistically significant and have meaningful relationships with the outcome variable.

### 4. Comment on the R^2 value.

#### The R² value is 0.656, meaning that approximately 65.6% of the variation in mental health score is explained by the model. This suggests a relatively strong model fit.